sist02.csl

```diff
@@ -8,199 +8,133 @@
-            <date-part name="month" form="numeric-leading-zeros" range-delimiter="-" strip-periods="true" suffix="-"/>
-            <date-part name="day" form="numeric-leading-zeros"/>
-          </date>
-        </group>
-      </if>
-    </choose>
+    <group delimiter=", ">
+      <text variable="URL" prefix=" "/>
+      <group>
+        <text term="accessed" text-case="lowercase" prefix="(" suffix=" "/>
+        <date variable="accessed" suffix=")">
+          <date-part name="year" range-delimiter="-" suffix="-"/>
+          <date-part name="month" form="numeric-leading-zeros" range-delimiter="-" strip-periods="true" suffix="-"/>
+          <date-part name="day" form="numeric-leading-zeros"/>
+        </date>
+      </group>
+    </group>
   </macro>
   <macro name="page">
     <choose>
-      <if type="article-journal" match="any">
-        <label plural="never" suffix=" " variable="page" form="short"/>
-        <text variable="page" suffix="."/>
-      </if>
-      <else-if type="book" match="any">
+      <if type="article-journal chapter" match="any">
+        <label plural="never" suffix=". " variable="page" form="short"/>
         <text variable="page"/>
-        <choose>
-          <if match="any" variable="collection-title">
-            <label plural="never" suffix=", " variable="page" form="short"/>
-          </if>
-          <else>
-            <label plural="never" suffix="." variable="page" form="short"/>
-          </else>
-        </choose>
+      </if>
+      <else-if type="book thesis" match="any">
+        <text variable="number-of-pages"/>
+        <label plural="never" variable="page" form="short"/>
       </else-if>
-      <else-if type="chapter" match="any">
-        <label plural="never" suffix=" " variable="page" form="short"/>
+      <else-if type="report" match="any">
         <choose>
-          <if match="any" variable="collection-title">
-            <text variable="page" suffix=", "/>
+          <if match="any" variable="container-title">
+            <label plural="never" suffix=". " variable="page" form="short"/>
+            <text variable="number-of-pages"/>
           </if>
           <else>
-            <text variable="page" suffix="."/>
+            <text variable="number-of-pages"/>
+            <label plural="never" variable="page" form="short"/>
           </else>
         </choose>
       </else-if>
-      <else-if type="paper-conference report" match="any">
+      <else>
         <choose>
           <if match="any" variable="container-title">
-            <label plural="never" suffix=" " variable="page" form="short"/>
-            <choose>
-              <if match="any" variable="collection-title">
-                <text variable="page" suffix=", "/>
-              </if>
-              <else>
-                <text variable="page" suffix="."/>
-              </else>
-            </choose>
+            <label plural="never" suffix=". " variable="page" form="short"/>
```

Comment:
Again, this might be an issue with Mendeley.  If having the period in the suffix does not do any harm as you said in your earlier message, will it be OK just leave my coding as it is for now?  I will raise the problem with the Mendeley team.

Shoji



From: Sebastian Karcher [mailto:notifications@github.com]
Sent: Monday, December 14, 2015 7:20 AM
To: citation-style-language/styles
Cc: Takahashi, Shoji (ELS-TOK)
Subject: Re: [styles] Update sist02.csl (#1828)


In sist02.csl<https://github.com/citation-style-language/styles/pull/1828#discussion_r47451359>:

>          <choose>

>            <if match="any" variable="container-title">

> -            <label plural="never" suffix=" " variable="page" form="short"/>

> -            <choose>

> -              <if match="any" variable="collection-title">

> -                <text variable="page" suffix=", "/>

> -              </if>

> -              <else>

> -                <text variable="page" suffix="."/>

> -              </else>

> -            </choose>

> +            <label plural="never" suffix=". " variable="page" form="short"/>

That's very odd:
https://github.com/citation-style-language/locales/blob/master/locales-ja-JP.xml#L172 has the period in the locale. You yourself fixed that more than a year ago.

(btw. -- you can't e-mail screenshots. You'd need to drag them to the message here)

—
Reply to this email directly or view it on GitHub<https://github.com/citation-style-language/styles/pull/1828/files#r47451359>.